lib/Basics/Thread.cpp

```diff
@@ -201,7 +201,7 @@ void Thread::shutdown() {
   LOG_TOPIC("93614", TRACE, Logger::THREADS) << "shutdown(" << _name << ")";
 
   beginShutdown();
```

Comment:
Should `beginShutdown();` call be moved  inside `if (_threadStructInitialized.exchange(false, std::memory_order_acquire)) {` ? As `shutdown()` may be called many times explicitly and one additional call from destructor. Before that there was only one call in `beginShutdown();` (as `shutdown()` was called from destructor)

Reply:
I don't think it is necessary to move the call to `beginShutdown` into the `if (_threadStructInitialized.exchange(false, ...)` section.

`beginShutdown` can be called multiple times, because it only does atomic compare-and-swap operations on the thread's `_state` variable. That should always be safe to do.

Reply:
`beginShutdown` implementations have to be idempotent, i.e., it must be possible to call them multiple times without changing the result beyond the initial application.
This was already the case before this PR, even though it was not stated explicitly anywhere; there are many cases where `beginShutdown` is called explicitly before a call to `shutdown`.